MyClassroom/Controllers/OrganizationsController.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using MyClassroom.Application.Services;
+using MyClassroom.MVC.Github;
+
+namespace MyClassroom.MVC.Controllers
+{
+    [Authorize]
+    public class OrganizationsController : Controller
+    {
+        private readonly IOrganizationService _organizationService;
+        private readonly IGitHubClientService _gitHubClientService;
+        private readonly ITokenService _tokenService;
+        private readonly GithubProductHeaderOptions _githubProductHeaderOptions;
+
+        public OrganizationsController(
+            IOrganizationService organizationService, 
+            IGitHubClientService gitHubClientService, 
+            ITokenService tokenService, 
+            IOptions<GithubProductHeaderOptions> githubProductHeaderOptions)
+        {
+            _organizationService = organizationService;
+            _gitHubClientService = gitHubClientService;
+            _tokenService = tokenService;
+            _githubProductHeaderOptions = githubProductHeaderOptions.Value;
+        }
+
+        public async Task<IActionResult> Index()
+        {
+            var session_id = HttpContext?.User?.FindFirst("session_id")?.Value;
+            if (Guid.TryParse(session_id, out Guid id))
+            {
+                var token = await _tokenService.GetTokenById(Guid.Parse(session_id));
```

Comment:
We can use the id variable here, instead of parsing again the session_id. I think I made the same mistake in HomeController.